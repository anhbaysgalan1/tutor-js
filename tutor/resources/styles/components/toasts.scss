.toast-notification {
  position: fixed;
  z-index: $zindex-modal;
  min-height: 50px;
  right: 20px;
  top: 20px;
  background-color: white;
  box-shadow: 0px 10px 20px rgba(0, 0, 0, 0.2);
  display: flex;
  align-items: center;
  max-width: 280px;

  .title {
    padding: 10px;
    font-weight: bold;
    background-color: #e0edd3;
    display: flex;
    align-content: center;
    justify-content: space-between;
    &::before {
      @include fa-icon();
      font-size: 26px;
      font-weight: normal;
      margin-right: 0.5rem;
    }
  }

  .body {
    padding: 10px;
  }

  .success {
    .title{
      color: #629030;
      &::before {
        content: $fa-var-check-circle;
      }
    }
  }

  .failure {
    .title {
      color: #C22032;
      background-color: #f5e9ea;
      &::before {
        content: $fa-var-exclamation-triangle;
      }
      .tutor-icon[type="close"] {
        color: $tutor-neutral-dark;
        margin-left: 2rem;
        align-self: flex-start;
      }
    }
    .body {
      button {
        padding-left: 0;
        font-size: 16px;
      }
    }
  }

  .scores {
    iframe {
      border: 0;
      height: 0;
      width: 0;
    }
  }

  .reload {
    button {
      margin-top: 1rem;
    }
  }
}

.lms-push-partial-failure {
  .modal-body {
    .controls {
      margin-top: 40px;
    }
    .actions {
      display: flex;
      justify-content: space-between;
      .btn {
        color: $tutor-neutral-dark;
        font-size: 16px;
        padding: 20px 0;
      }
    }
    table th {
      vertical-align: bottom;
      &.average {
        display: flex;
        justify-content: space-between;
        .tutor-icon {
          border: 1px solid $tutor-neutral;
          padding: 4px;
        }
      }
    }
    .panel {
      border: 0;
      box-shadow: none;
      margin: 0;
      .panel-body {
        max-height: 320px;
        overflow: auto;
      }
    }
  }
}
