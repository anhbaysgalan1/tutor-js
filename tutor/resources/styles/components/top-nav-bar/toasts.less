.background-job-toast {
  position: fixed;
  min-height: 50px;
  right: 20px;
  top: 20px;
  background-color: white;
  box-shadow: 0px 10px 20px rgba(0, 0, 0, 0.2);
  display: flex;
  align-items: center;

  .success {
    margin: 10px;
    &::before {
      .fa-icon();
      font-size: 26px;
      font-weight: normal;
      content: @fa-var-check;
      color: #63A524;
      margin-right: 0.5rem;
    }
  }

  .failure {
    // color: #b33238;
    display: flex;
    align-items: flex-start;
    flex-direction: column;
    .heading {
      display: flex;
      font-weight: 700;
      font-style: normal;
      color: #C22032;
      background-color: #f5e9ea;
      padding: 20px;
      .tutor-icon[type="close"] {
        color: @tutor-neutral-dark;
        font-size: 20px;
        margin-left: 1rem;
        align-self: flex-start;
        margin-top: -10px;
        margin-right: -10px;
      }
      &::before {
        .fa-icon();
        font-size: 26px;
        font-weight: normal;
        content: @fa-var-exclamation-triangle;
//        color: #63A524;
        margin-right: 0.5rem;
      }
    }
    button {
      margin: 10px 0;
      font-size: 16px;
    }
  }

  .scores {
    iframe {
      border: 0;
      height: 0;
      width: 0;
    }
  }
}
