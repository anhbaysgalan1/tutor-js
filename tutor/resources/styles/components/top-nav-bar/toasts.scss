.background-job-toast {
  position: fixed;
  min-height: 50px;
  right: 20px;
  top: 20px;
  background-color: white;
  box-shadow: 0px 10px 20px rgba(0, 0, 0, 0.2);
  display: flex;
  align-items: center;

  .success {
    margin: 10px;
    &::before {
      @include fa-icon();
      font-size: 26px;
      font-weight: normal;
      content: $fa-var-check;
      color: #63A524;
      margin-right: 0.5rem;
    }
  }

  .failure {
    // color: #b33238;
    display: flex;
    align-items: flex-start;
    flex-direction: column;
    .heading {
      display: flex;
      font-weight: 700;
      font-style: normal;
      color: #C22032;
      background-color: #f5e9ea;
      padding: 20px;
      .tutor-icon[type="close"] {
        color: $tutor-neutral-dark;
        font-size: 20px;
        margin-left: 1rem;
        align-self: flex-start;
        margin-top: -10px;
        margin-right: -10px;
      }
      &::before {
        @include fa-icon();
        font-size: 26px;
        font-weight: normal;
        content: $fa-var-exclamation-triangle;
//        color: #63A524;
        margin-right: 0.5rem;
      }
    }
    button {
      margin: 10px 0;
      font-size: 16px;
    }
  }

  .scores {
    iframe {
      border: 0;
      height: 0;
      width: 0;
    }
  }

}

.lms-push-partial-failure {
  .modal-body {
    .actions {
      display: flex;
      justify-content: space-between;
      .btn {
        color: $tutor-neutral-dark;
        font-size: 16px;
        padding: 20px 0;
      }
    }
    .panel {
      border: 0;
      box-shadow: none;
      margin: 0;
      .panel-body {
        max-height: 320px;
        overflow: auto;
      }
    }
  }
}
