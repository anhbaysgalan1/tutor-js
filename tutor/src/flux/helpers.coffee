_ = require 'underscore'
flux = require 'flux-react'

LOADING = 'loading'
LOADED  = 'loaded'
FAILED  = 'failed'
SAVING  = 'saving'
DELETING = 'deleting'
DELETED = 'deleted'

STATES =
  LOADING : LOADING
  LOADED  : LOADED
  FAILED  : FAILED
  SAVING  : SAVING
  DELETING: DELETING
  DELETED : DELETED

idCounter = 0
CREATE_KEY = -> "_CREATING_#{idCounter++}"

isNew = (id) -> /_CREATING_/.test(id)

CrudConfig = ->
  {
    _asyncStatus: {}
    _local: {}
    _changed: {}
    _errors: {}
    _reload: {}

    # If the specific type needs to do something else to the object:
    # _loaded : (obj, id) ->
    # _saved : (obj, id) ->
    # _reset : ->

    reset: ->
      @_asyncStatus = {}
      @_local = {}
      @_changed = {}
      @_errors = {}
      @_reload = {}

      @_reset?()
      @emitChange()

    dontReload: (id) ->
      @_asyncStatus[id] is LOADED and @_HACK_DO_NOT_RELOAD

    FAILED: (status, msg, id, args...) ->
      @_asyncStatus[id] = FAILED
      @_errors[id] = msg
      @_failed?({status, msg}, id, args...)

      unless status is 0 # indicates network failure
        delete @_local[id]
        @emitChange()

    load: (id) ->
      @_load(id)

    # allow aliasing load without triggering api call.
    _load: (id) ->
      # Add a shortcut for unit testing
      return if @dontReload(id)
      @_reload[id] = false
      @_asyncStatus[id] = LOADING
      @emitChange()

    loaded: (obj, id, args...) ->
      # id = obj.id
      @_asyncStatus[id] = LOADED

      if obj
        # If the specific type needs to do something else to the object:
        @_local[id] = @_loaded?(obj, id, args...) or obj

      @emitChange()

    save: (id, obj) ->
      # Note: id could be isNew()
      @_asyncStatus[id] = SAVING
      @emitChange()

    saved: (result, id, args...) ->
      # id = result.id
      @_asyncStatus[id] = LOADED # TODO: Maybe make this SAVED

      # If the specific type needs to do something else to the object:
      obj = @_saved?(result, id, args...)
      result = obj if obj

      if result
        @_local[id] = result
        @_local[result.id] = result
        delete @_changed[result.id]
      else
        console.warn('API WARN: Server did not return JSON after saving. Patching locally')
        # Merge all the local changes into the new local object
        @_local[id] = _.extend(@_local[id], @_changed[id])
      delete @_changed[id]
      delete @_errors[id]
      # If the specific type needs to do something else to the object:
      @emitChange()

    create: (localId, attributes = {}) ->
      throw new Error('BUG: MUST provide a local id') unless isNew(localId)
      @_local[localId] = {}
      @_changed[localId] = attributes
      @_asyncStatus[localId] = LOADED

    created: (result, localId, args...) ->
      @_local[localId] = result # HACK: So react component can still manipulate the same object
      @_local[result.id] = result
      @_asyncStatus[localId] = LOADED
      @_asyncStatus[result.id] = LOADED
      @emitChange()
      @_created?(result, localId, args...)

    _change: (id, obj) ->
      @_changed[id] ?= {}
      _.extend(@_changed[id], obj)
      @emitChange()

    _save: (id) ->
      @_asyncStatus[id] = SAVING

    delete: (id) ->
      @_asyncStatus[id] = DELETING
      @_delete?(id)

    deleted: (result, id, args...) ->
      @_asyncStatus[id] = DELETED
      delete @_local[id]
      @emitChange()
      @_deleted?(result, id, args...)

    clearChanged: (id) ->
      delete @_changed[id]

    HACK_DO_NOT_RELOAD: (bool) -> @_HACK_DO_NOT_RELOAD = bool

    # Keep this here so other exports method have access to it
    _get: (id) ->
      val = @_local[id]
      return null unless val or @_asyncStatus[id] is SAVING
      # Scores stores an Array unlike most other stores
      if _.isArray(val)
        val
      else
        _.extend({}, val, @_changed[id])

    exports:
      isUnknown: (id) -> not @_asyncStatus[id]
      isLoading: (id) -> @_asyncStatus[id] is LOADING
      isLoaded: (id) -> @_asyncStatus[id] is LOADED
      isDeleting: (id) -> @_asyncStatus[id] is DELETING
      isSaving: (id) -> @_asyncStatus[id] is SAVING
      isFailed: (id) -> @_asyncStatus[id] is FAILED
      getAsyncStatus: (id) -> @_asyncStatus[id]
      get: (id) -> @_get(id)
      isChanged: (id) -> not _.isEmpty(@_changed[id])
      getChanged: (id) -> @_changed[id] or {}
      freshLocalId: -> CREATE_KEY()
      isNew: (id) -> isNew(id)
      reload: (id) -> @_reload[id]
      getError: (id) -> @_errors[id]

  }

# Helper for creating a simple store for actions
makeSimpleStore = (storeConfig) ->

  actionsConfig = _.omit storeConfig, (value, key) ->
    typeof value isnt 'function' or key is 'exports'

  actionsConfig = _.keys(actionsConfig)
  actions = flux.createActions(actionsConfig)
  storeConfig.actions = _.values(actions)
  store = flux.createStore(storeConfig)
  {actions, store}


extendConfig = (newConfig, origConfig) ->
  newConfig.exports ?= {}
  _.defaults(newConfig, origConfig)
  _.defaults(newConfig.exports, origConfig.exports)
  newConfig


makeStandardStore = (name, storeConfig) ->
  extendConfig(storeConfig, new CrudConfig())
  {actions, store} = makeSimpleStore(storeConfig)
  {"#{name}Actions":actions, "#{name}Store":store}


module.exports = {CrudConfig, makeSimpleStore, extendConfig, makeStandardStore, isNew, STATES}
