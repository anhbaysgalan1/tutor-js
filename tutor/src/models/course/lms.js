import { action, observable, computed } from 'mobx';
import {
  BaseModel, identifiedBy, identifier, session, belongsTo,
} from '../base';
import { UiSettings } from 'shared';

const LMS_VENDOR = 'lmsv';

@identifiedBy('course/lms')
export default class CourseLMS extends BaseModel {

  @observable course;

  @identifier id;
  @session key;
  @session secret;
  @session launch_url;
  @session configuration_url;
  @session xml;

  @session({ type: 'date' }) created_at;
  @session({ type: 'date' }) updated_at;

  @belongsTo({ model: 'course' }) course;

  @computed get vendor() {
    return UiSettings.get(LMS_VENDOR, this.course.id) || 'blackboard';
  }

  @action.bound setVendor(vendor) {
    console.log(vendor)
    return UiSettings.set(LMS_VENDOR, this.course.id, vendor);
  }

  // called by API
  pushScores() {}
  fetch() { }

}
