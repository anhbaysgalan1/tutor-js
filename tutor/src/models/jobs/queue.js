import { observable } from 'mobx';
import { readonly } from 'core-decorators';
import { createCollection } from 'mobx-decorated-models';
import { uniqueId } from 'lodash';

import {
  BaseModel, identifiedBy, session,
} from '../base';

@identifiedBy('jobs/queue')
export class JobCompletion extends BaseModel {

  @readonly id = uniqueId();

  @session succeeded;
  @session type;
  @session({ type: 'object' }) info;
  @session({ type: 'object' }) component;

}


const Completed = createCollection({ model: JobCompletion });

export { Completed };
