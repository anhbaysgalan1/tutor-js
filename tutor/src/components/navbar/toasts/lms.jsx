import React from 'react';
import { observer } from 'mobx-react';
import { observable, action, computed } from 'mobx';
import { Button, Panel, Table } from 'react-bootstrap';
import { isEmpty } from 'lodash';
import Icon from '../../icon';
import WarningModal from '../../warning-modal';
import NewTabLink from '../../new-tab-link';
import { JobCompletion } from '../../../models/jobs/queue';
import S from '../../../helpers/string';
import { downloadData, arrayToCSV } from '../../../helpers/download-data';

@observer
class Errors extends React.Component {

  static propTypes = {
    job: React.PropTypes.instanceOf(JobCompletion).isRequired,
    footer: React.PropTypes.node.isRequired,
  }

  @observable displayInfo = false;

  @action.bound toggleInfo() { this.displayInfo = !this.displayInfo; }

  @computed get errorData() {
    return this.props.job.info.errors.map((e) =>
      [
        e.student_identifier,
        e.student_name,
        S.numberWithTwoDecimalPlaces(e.score * 100),
      ]);
  }

  @action.bound startDownload() {
    const rows = [
      [ 'Student ID', 'Name', 'CourseAverage' ],
    ].concat(this.errorData);
    downloadData(arrayToCSV(rows), 'failed-scores.csv', 'text/csv');
  }

  render() {
    const { displayInfo, props: { footer, job } } = this;
    return (
      <WarningModal
        className="lms-push-partial-failure"
        backdrop={false}
        title="Some scores not sent"
        footer={footer}
      >
        <p>
          Course averages for {job.info.errors.length} could not be
          sent successfully to your LMS.  There may be an issue with
          your LMS, or something may have happened when students enrolled.
        </p>
        <div className="actions">
          <Button bsStyle="link" onClick={this.toggleInfo}>
            {displayInfo ? 'Hide' : 'Show'} scores not sent
          </Button>
          <Button bsStyle="link" onClick={this.startDownload}>
            <Icon type="download" /> Download failures
          </Button>
        </div>
        <Panel collapsible expanded={this.displayInfo}>
          <Table>
            <thead>
              <tr>
                <td>Student ID</td>
                <td>Name</td>
                <td>Course Average</td>
              </tr>
            </thead>
            <tbody>
              {this.errorData.map(([id, name, score]) => (
                <tr>
                  <td>{id}</td>
                  <td>{name}</td>
                  <td>{score}</td>
                </tr>
              ))}
            </tbody>
          </Table>
        </Panel>
      </WarningModal>
    );
  }
}

const renderNoScores = (footer) => (
  <WarningModal
    backdrop={false}
    title="No scores to send"
    footer={footer}
  >
    <p>
      This course has no scores to send, either because there are no
      students enrolled or no assignments have been created.
    </p>
    <NewTabLink
      to="http://4tk3oi.axshare.com/salesforce_support_page_results.html#choose_support_article=All&CSUM=1"
    >
      Troubleshoot sending scores to your LMS
    </NewTabLink>

  </WarningModal>
);

const renderFailedToSend = (footer) => (
  <WarningModal
    backdrop={false}
    title="Scores not sent"
    footer={footer}
  >
    <p>
      Your course averages could not be sent, either due to a network timeout or a problem in your LMS. Try again on the Scores page.
      If this problem persists, you may need to perform extra steps in your LMS. This support article shows you how:
    </p>
    <div>
      <p>
        <NewTabLink
          to="http://4tk3oi.axshare.com/salesforce_support_page_results.html#choose_support_article=All&CSUM=1"
        >
          send course averages to your LMS
        </NewTabLink>
      </p>
    </div>
  </WarningModal>
);

@observer
export class Failure extends React.Component {

  static propTypes = {
    dismiss: React.PropTypes.func.isRequired,
    job: React.PropTypes.instanceOf(JobCompletion).isRequired,
  }

  @observable showDetails = false;
  @action.bound onShowDetails() { this.showDetails = true; }

  render() {
    const { job } = this.props;

    const { info: { errors, data: {
      num_callbacks,
    } } } = job;

    if (this.showDetails) {
      const footer = <Button onClick={this.props.dismiss}>Close</Button>;
      if (!isEmpty(errors)) {
        return <Errors {...{ job, footer }} />;
      } else if (num_callbacks) {
        return renderFailedToSend(footer);
      } else {
        return renderNoScores(footer);
      }
    }

    return (
      <div className="toast scores failure">
        <div className="heading">
          {num_callbacks ? 'Scores not sent' : 'Some scores not sent'}
          <Icon type="close" onClick={this.props.dismiss} />
        </div>
        <Button bsStyle="link" onClick={this.onShowDetails}>Details</Button>
      </div>
    );
  }
}

export function Success() {
  return (
    <div className="toast success">
      Scores successfully sent
    </div>
  );
}
