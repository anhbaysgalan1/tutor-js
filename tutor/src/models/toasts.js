import { readonly } from 'core-decorators';
import { createCollection } from 'mobx-decorated-models';
import { uniqueId } from 'lodash';
import invariant from 'invariant';
import {  computed } from 'mobx';
import {
  BaseModel, identifiedBy, session,
} from 'shared/model';

import * as lms from '../components/toasts/lms';
import * as scores from '../components/toasts/scores';
import Reload from '../components/toasts/reload';

const JobToasts = { lms, scores };

const Handlers  = {
  job(toast) {
    invariant(['ok', 'failed'].includes(toast.status), 'job status must be ok or failed');
    return JobToasts[toast.type][toast.status == 'ok' ? 'Success' : 'Failure'];
  },
  reload() {
    return Reload;
  },
};

@identifiedBy('toast')
export class ToastModel extends BaseModel {

  @readonly id = uniqueId('toast-');
  @session handler;
  @session status;
  @session type;
  @session({ type: 'object' }) info;

  @computed get isOk() {
    return 'ok' === this.status;
  }

  @computed get component() {
    const handler = Handlers[this.handler];
    invariant(handler, `Handler type for '${this.handler}' was not found`);
    return handler(this);
  }
}


const ToastsStore = createCollection({ model: ToastModel });

export default ToastsStore;
