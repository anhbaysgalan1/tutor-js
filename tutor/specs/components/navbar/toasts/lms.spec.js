import { Success, LMSErrors, Failure } from '../../../../src/components/navbar/toasts/lms';
import EnzyeContext from '../../helpers/enzyme-context';
import { SnapShot, Wrapper } from '../../helpers/component-testing';
import { portalContents as PC } from '../../../helpers/portals';

import  { JobCompletion, Completed } from '../../../../src/models/jobs/queue';
jest.useFakeTimers();

describe('Background job toasts', () => {
  let toast;
  let job;
  let props;

  beforeEach(() => {

    job = new JobCompletion({
      succeeded: true,
      type: 'lms',
      info: {
        errors: [
          { student_identifier: '1234', student_name: 'Bob', score: 0.123 },
        ],
        data: {
          num_callbacks: 0,
        },
      },
    });
    props = { job, dismiss: jest.fn, footer: <span /> };
    Completed.push(job);
  });

  afterEach(() => Completed.clear());

  it('pluralizes error count', () => {
    toast = mount(<LMSErrors {...props} />, EnzyeContext.build());
    expect(PC(toast).textContent).toContain(
      'Course averages for 1 student could not be sent'
    );
    job.info.errors.push({
      student_identifier: '4321', student_name: 'Jane', score: 0.923,
    });
    toast = mount(<LMSErrors {...props} />, EnzyeContext.build());
    expect(PC(toast).textContent).toContain(
      'Course averages for 2 students could not be sent'
    );
    expect(PC(toast).innerHTML).toMatchSnapshot();
  });

  it('toggles error list', () => {
    toast = mount(<LMSErrors {...props} />, EnzyeContext.build());
    expect(
      PC(toast).querySelector('.btn.toggle').textContent
    ).toContain('Show');
    PC(toast).querySelector('.btn.toggle').click();
    const body = PC(toast);
    expect(
      body.querySelector('.btn.toggle').textContent
    ).toContain('Hide');
    expect(body.querySelector('tbody').textContent).toContain(
      'Bob'
    );
    expect(body.querySelector('tbody').textContent).toContain(
      '12.3'
    );
    expect(PC(toast).innerHTML).toMatchSnapshot();
  });

});
