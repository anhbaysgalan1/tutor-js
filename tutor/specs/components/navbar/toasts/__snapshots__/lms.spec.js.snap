// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Background job toasts pluralizes error count 1`] = `"<div data-reactroot=\\"\\" role=\\"dialog\\"><div role=\\"dialog\\" tabindex=\\"-1\\" class=\\"warning lms-push-partial-failure fade in modal\\" style=\\"display: block; padding-right: 0px;\\"><div class=\\"modal-dialog\\"><div class=\\"modal-content\\" role=\\"document\\"><div class=\\"warning modal-header\\"><i type=\\"exclamation-triangle\\" role=\\"presentation\\" class=\\"tutor-icon fa fa-exclamation-triangle\\"></i><!-- react-text: 7 -->Some scores not sent<!-- /react-text --></div><div class=\\"modal-body\\"><div><p><!-- react-text: 11 -->Course averages for <!-- /react-text --><!-- react-text: 12 -->2 students<!-- /react-text --><!-- react-text: 13 --> could not be sent successfully to your LMS.  There may be an issue with your LMS, or something may have happened when students enrolled.<!-- /react-text --></p><a target=\\"_blank\\" tabindex=\\"0\\">Troubleshoot sending scores to your LMS</a><div class=\\"controls\\"><button type=\\"button\\" class=\\"btn btn-primary\\">View those scores</button><button type=\\"button\\" class=\\"btn btn-default\\">Close</button></div></div></div></div></div></div></div>"`;

exports[`Background job toasts toggles error list 1`] = `"<div data-reactroot=\\"\\" role=\\"dialog\\"><div role=\\"dialog\\" tabindex=\\"-1\\" class=\\"warning lms-push-partial-failure fade in modal\\" style=\\"display: block; padding-right: 0px;\\"><div class=\\"modal-dialog\\"><div class=\\"modal-content\\" role=\\"document\\"><div class=\\"warning modal-header\\"><i type=\\"exclamation-triangle\\" role=\\"presentation\\" class=\\"tutor-icon fa fa-exclamation-triangle\\"></i><!-- react-text: 7 -->Some scores not sent<!-- /react-text --></div><div class=\\"modal-body\\"><div><p>SCORES NOT SENT</p><table class=\\"table\\"><thead><tr><th>Student ID</th><th>Name</th><th class=\\"average\\"><!-- react-text: 24 -->Course Average<!-- /react-text --><i type=\\"download\\" role=\\"button\\" class=\\"tutor-icon fa fa-download clickable\\"></i></th></tr></thead><tbody><tr><td>1234</td><td>Bob</td><td><!-- react-text: 31 -->12<!-- /react-text --><!-- react-text: 32 -->%<!-- /react-text --></td></tr></tbody></table><div class=\\"controls\\"><button type=\\"button\\" class=\\"dark btn btn-default\\">Back</button><button type=\\"button\\" class=\\"btn btn-default\\">Close</button></div></div></div></div></div></div></div>"`;
